agent/engine/dependencygraph/graph.go

```diff
@@ -304,41 +311,55 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 	dependsOnContainerKnownStatus := dependsOnContainer.GetKnownStatus()
 
 	switch dependsOnStatus {
-
-	case "START":
-		// The 'target' container doesn't desire to be moved to 'Created' or the 'steady' state.
-		// which is not allowed.
-		if targetDesiredStatus != apicontainerstatus.ContainerCreated && targetDesiredStatus != apicontainerstatus.ContainerRunning {
-			return false
-		}
+	case StartCondition:
 		// The 'target' container desires to be moved to 'Created' or the 'steady' state.
 		// Allow this only if the known status of the dependency container state is already started
 		// i.e it's state is any of 'Created', 'steady state' or 'Stopped'
 		return dependsOnContainerKnownStatus == apicontainerstatus.ContainerCreated ||
 			dependsOnContainerKnownStatus == apicontainerstatus.ContainerStopped ||
 			dependsOnContainerKnownStatus == dependsOnContainer.GetSteadyStateStatus()
 
-	case "RUNNING":
-
+	case RunningCondition:
 		if targetDesiredStatus == apicontainerstatus.ContainerCreated {
 			// The 'target' container desires to be moved to 'Created' state.
 			// Allow this only if the known status of the linked container is
 			// 'Created' or if the dependency container is in 'steady state'
 			return dependsOnContainerKnownStatus == apicontainerstatus.ContainerCreated || dependsOnContainer.IsKnownSteadyState()
-
 		} else if targetDesiredStatus == target.GetSteadyStateStatus() {
 			// The 'target' container desires to be moved to its 'steady' state.
 			// Allow this only if the dependency container is in 'steady state' as well
 			return dependsOnContainer.IsKnownSteadyState()
-
 		}
 		return false
 
+	case SuccessCondition:
+		// The 'target' container desires to be moved to 'Created' or the 'steady' state.
+		// Allow this only if the known status of the dependency container state is stopped with an exit code of 0
+		return dependsOnContainerKnownStatus == apicontainerstatus.ContainerStopped && *dependsOnContainer.GetKnownExitCode() == 0
+
+	case CompleteCondition:
+		// The 'target' container desires to be moved to 'Created' or the 'steady' state.
+		// Allow this only if the known status of the dependency container state is stopped with any exit code
+		return dependsOnContainerKnownStatus == apicontainerstatus.ContainerStopped && dependsOnContainer.GetKnownExitCode() != nil
+
+	case HealthyCondition:
```

Comment:
are we verifying if health check is enabled for the `dependsOnContainer` if dependent status is `HEALTHY`?

Reply:
Thanks Sharanya for pointing this out. I have included this check now.

Reply:
let's check this on the service side as well